blockly-core/core/ui/function_editor.js

```diff
@@ -798,6 +799,20 @@ Blockly.FunctionEditor.prototype.addDeleteButton_ = function () {
       this.onDeletePressed.bind(this));
 };
 
+/**
+ * Show or hide delete button. If none exists and it should be shown, create
+ * it.
+ */
+Blockly.FunctionEditor.prototype.setDeleteButtonVisible_ = function(visible) {
+  if (this.deleteButton_) {
+    this.deleteButton_.setVisible(visible);
+  } else if (visible) {
+    this.hasDeleteButton = true;
```

Comment:
Ah, I see it's used by `addDeleteButton`. Is this set anywhere else? Why is it not set by `addDeleteButton`?